### PR TITLE
Remove 'http' entries from CSP.

### PR DESCRIPTION
**Pull request by akatsoulas**

### CODE REVIEW

**Comment by johngian** [comment]
Doubled checked there are no references to those entries. The issue is going to be in our local envs that will not be able to load assets that are fetched depending in the site url (eg starting with "//")

**Comment by akatsoulas** [comment]
We could add these entries in the `local.py-dist`. Thoughts?

**Comment by johngian** [comment]
sounds good

**Comment by akatsoulas** [comment]
@johngian pr updated

**Comment by johngian** [comment]
``
  File "/code/mozillians/settings/local.py", line 90, in <module>
    base.CSP_DEFAULT_SRC += ('http://*.mapbox.com',)
NameError: name 'base' is not defined
``

**Comment by akatsoulas** [comment]
Ermmm, my bad, pr updated.

File: mozillians/settings/local.py-dist [hunk 1]
```diff
@@ -53,4 +53,17 @@ CACHES = {
         'LOCATION': 'unique-snowflake'
     }
 }
-BROWSERID_AUDIENCES = None
\ No newline at end of file
+BROWSERID_AUDIENCES = None
+base.CSP_DEFAULT_SRC += ('http://*.mapbox.com',)
```
Comment (johngian) [inline mozillians/settings/local.py-dist]: You don't really override the values here. Try something like:
```

CSR_DEFAULT_SRC = base.CSP_DEFAULT_SRC + ('http://*.mapbox.com',)
CSP_FONT_SRC = base.CSP_FONT_SRC +('http://*.mozilla.net',
                                   'http://*.mozilla.org',
                                   'http://mozorg.cdn.mozilla.net',)
CSP_IMG_SRC = base.CSP_IMG_SRC + ('http://*.mozilla.net',
                                  'http://*.mozilla.org',
                                  'http://*mapbox.com',)
CSP_SCRIPT_SCR = base.CSP_SCRIPT_SRC + ('http://*.mozilla.net',
                                        'http://*.mozilla.org',
                                        'http://*mapbox.com',)
CSP_STYLE_SRC = base.CSP_STYLE_SRC + ('http://*.mozilla.net',
                                      'http://*.mozilla.org',
                                      'http://*mapbox.com',)
```

File: mozillians/settings/local.py-docker-dist [hunk 2]
```diff
@@ -85,4 +86,18 @@ CACHES = {
     }
 }
 
-BROWSERID_AUDIENCES = None
\ No newline at end of file
+BROWSERID_AUDIENCES = None
+
+base.CSP_DEFAULT_SRC += ('http://*.mapbox.com',)
```
Comment (johngian) [inline mozillians/settings/local.py-docker-dist]: Same here.

**Comment by johngian** [comment]
r+

**Comment by glogiotatidis** [comment]
Just curious: How do you handle `//example.com/foo.js` links during development where you don't have `https` and therefore the example translates to `http://example.com/foo.js`

**Comment by johngian** [comment]
This should be addressed after overriding CSP related configuration entries in ``settings/local.py-{dist, docker-dist}``.

**Comment by glogiotatidis** [comment]
ic, that's tricky. Thanks for the answer!

**Comment by akatsoulas** [comment]
@glogiotatidis can you please elaborate more about the tricky part? Do you have any ideas for a cleaner solution?

**Comment by glogiotatidis** [comment]
I consider that tricky because you'll not face CSP errors during development and the possibilities to land code that doesn't work on stage is higher. Unless you actually maintain the `http` version of CSP rules on your local file.

I prefer to keep `http` the CSP rules in place, since we run everything under `https` on production I don't see how they can be a problem.

**Comment by akatsoulas** [comment]
mozillians.org was allowing `http` on prod  thus this change. Thanks for the explanation! :beer:

**Comment by johngian** [comment]
FYI this was implemented after some security suggestions not to allow CSP content using ``http``. That's why the explicit CSP rule.

**Pull request merged**